projects/plutus/src/current_sense.c

```diff
@@ -5,52 +5,45 @@
 #include "critical_section.h"
 #include "wait.h"
 
-// Must be less than 128, or overflow may occur
-#define CURRENT_SENSE_RESET_SAMPLES 10
-
-static void prv_reset_samples(int32_t *value, void *context) {
-  CurrentSenseStorage *storage = (CurrentSenseStorage *)context;
-  if (storage->samples < CURRENT_SENSE_RESET_SAMPLES) {
-    storage->data->zero_point.voltage += *value;
-    storage->samples++;
-  }
-}
-
 static void prv_calculate_current(int32_t *value, void *context) {
   CurrentSenseStorage *storage = (CurrentSenseStorage *)context;
 
   // Formula for calculating calibrated current. Draws slope between given calibrated
   // points, and uses the result as well as the voltage offset to calculate current
-  storage->current_value = storage->data->max_point.current *
+  storage->value.voltage = *value;
+  storage->value.current = storage->data->max_point.current *
                            (*value - storage->data->zero_point.voltage) /
                            (storage->data->max_point.voltage - storage->data->zero_point.voltage);
+
   if (storage->callback != NULL) {
-    storage->callback(storage->current_value, storage->context);
+    storage->callback(storage->value.current, storage->context);
   }
 }
 
 StatusCode current_sense_init(CurrentSenseStorage *storage, CurrentSenseCalibrationData *data,
-                              LtcAdcStorage *adc_storage, LtcAdcSettings *settings) {
-  if (storage == NULL || adc_storage == NULL || settings == NULL) {
+                              LtcAdcSettings *settings) {
+  if (storage == NULL || settings == NULL) {
     return status_code(STATUS_CODE_UNINITIALIZED);
   }
 
   // Initialize ADC and start periodic polling
-  storage->adc_storage = adc_storage;
+  storage->adc_storage = (LtcAdcStorage){ 0 };
```

Comment:
This is some really funky syntax. Why are you doing this? `ltc_adc_init` can handle uninitialized storage.

If you wanted to actually clear it memsetting it would be cleaner.

Reply:
Removed